src/codec.rs

```diff
@@ -205,6 +216,35 @@ impl<R: std::io::Read + std::io::Seek> Input for IoReader<R> {
 	fn read(&mut self, into: &mut [u8]) -> Result<(), Error> {
 		self.0.read_exact(into).map_err(Into::into)
 	}
+
+	fn skip(&mut self, len: usize) -> Result<(), Error> {
+		use std::convert::TryInto;
+		use std::io::SeekFrom;
+
+		let old_pos = self.0.seek(SeekFrom::Current(0))?;
+		let io_len = self.0.seek(SeekFrom::End(0))?;
+		let new_pos = old_pos.checked_add(len as u64)
+			.ok_or(Error::from("len can't be skipped, seek from start don't fit into u64"))?;
+
+		if new_pos <= io_len {
+			self.0.seek(SeekFrom::Start(new_pos))?;
+		} else {
+			self.0.seek(SeekFrom::Start(io_len))?;
```

Comment:
As far as I can see, Seek end position is not final, it can change over time, in substrate I expect our implementation to do so, seek::End(0) will return the temporar end but can grow when we read on it.
I open this issue for fixing IoReader in this regard: https://github.com/paritytech/parity-scale-codec/pull/212

Reply:
as follow up of #212 skip is not optimized anymore for IoReader as `Read` doesn't provide enough to do so.